mitodl/open-discussions
# Differentiate between user lists and learning paths in the search index

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #2322

#### What's this PR do?
- Sets the `object_type` of the `ESUserListSerializer` to the `list_type` of the `UserList`
- Modifies the front-end to display facet choices for both.

#### How should this be manually tested?
- Create some user lists and learning paths, some private, some public
- Recreate the search index if you already have the above
- Go to `http://od.odl.local:8063/courses/search` - both `Learning Paths` and `User Lists` should show up under the `Learning Resource` facet, with accurate counts, and work as expected when checked/unchecked.  Private lists and paths should only be returned if you are logged in as the author of those lists.


#### Screenshots (if appropriate)

![Screen Shot 2019-10-24 at 1 04 27 PM](https://user-images.githubusercontent.com/187676/67508381-e8868700-f65e-11e9-8168-3b89c54c4492.png)
